File: frontend/types/seller/payment-method.ts
export type PaymentMethodType = "CARD_TO_CARD" | "CRYPTO" | "EXTERNAL_GATEWAY";

export type PaymentMethod = {
  id: number;
  store_id: number;
  type: PaymentMethodType;
  display_name: string;
  card_number: string | null;
  wallet_address: string | null;
  external_url: string | null;
  owner_name: string | null;
  instructions: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
};

export type PaymentMethodCreate = {
  type: PaymentMethodType;
  display_name: string;
  card_number?: string | null;
  wallet_address?: string | null;
  external_url?: string | null;
  owner_name?: string | null;
  instructions?: string | null;
  is_active?: boolean;
};

export type PaymentMethodUpdate = {
  type?: PaymentMethodType;
  display_name?: string;
  card_number?: string | null;
  wallet_address?: string | null;
  external_url?: string | null;
  owner_name?: string | null;
  instructions?: string | null;
  is_active?: boolean;
};

File: backend/app/models/payment_method.py
from __future__ import annotations

from typing import TYPE_CHECKING, Optional

from sqlalchemy import Boolean, Enum, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base
from app.models.enums import PaymentMethodType
from app.models.mixins import TimestampMixin

if TYPE_CHECKING:
    from app.models.order import Order
    from app.models.store import Store


class PaymentMethod(TimestampMixin, Base):
    __tablename__ = "payment_methods"

    id: Mapped[int] = mapped_column(primary_key=True)
    store_id: Mapped[int] = mapped_column(
        ForeignKey("stores.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    type: Mapped[PaymentMethodType] = mapped_column(
        Enum(PaymentMethodType, native_enum=False),
        nullable=False,
    )
    display_name: Mapped[str] = mapped_column(String(255), nullable=False)
    card_number: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    wallet_address: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    external_url: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    owner_name: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    instructions: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    store: Mapped["Store"] = relationship("Store", back_populates="payment_methods")
    orders: Mapped[list["Order"]] = relationship("Order", back_populates="payment_method")
